app/views/unattended/provisioning_templates/snippet/redhat_register.erb

```diff
@@ -162,6 +166,13 @@ description: |
     subscription-manager attach --auto
 
   <% end -%>
+
+  <% if host_param_true?('subscription_manager_refresh', false) -%>
+  subscription-manager refresh <% if host_param_true?('subscription_manager_refresh_force', false) %> --force<% end %>
+  <% end -%>
+
+  subscription-manager status
```

Comment:
I'd also leave the `status` call inside the conditional. We don't want an unintended performance hit especially with hundreds of concurrent registrations. `status` does make API requests to Foreman.